tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -3795,30 +3592,30 @@ Status FragmentMetadata::load_footer(
     return Status::Ok();
 
   Buffer buff;
-  shared_ptr<ConstBuffer> cbuff = nullptr;
-  if (f_buff == nullptr) {
+  shared_ptr<Deserializer> deserializer = nullptr;
+  if (f_tile == nullptr) {
     has_consolidated_footer_ = false;
     RETURN_NOT_OK(read_file_footer(&buff, &footer_offset_, &footer_size_));
```

Comment:
I think read_file_footer can be converted to use a tile instead of a buffer.

Reply:
Also, you might be able to call get_footer_offset_and_size, when f_tile == nullptr to get the size, that way you can create a Tile object outside of an if statement, and get rid of the use of shared_ptr fully for the deserializer.

Reply:
addressed differently than suggested, but shared_ptr no longer used tho
deserializer is constructed outside of if